packages/webamp/js/components/WinampButton.tsx

```diff
@@ -28,56 +32,62 @@ type Props = DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>;
  */
 export default function WinampButton(props: Props): JSX.Element {
   const [active, setActive] = useState(false);
-  const originalOnMouseDown = props.onMouseDown;
+  const originalOnPointerDown = props.onPointerDown;
+  const { requireClicksOriginateLocally } = props;
 
-  const onMouseDown = useCallback(
+  const onPointerDown = useCallback(
     (e) => {
-      if (originalOnMouseDown != null) {
-        originalOnMouseDown(e);
+      if (originalOnPointerDown != null) {
+        originalOnPointerDown(e);
+      }
+      // Release the pointer capture
+      // https://w3c.github.io/pointerevents/#implicit-pointer-capture
+      // https://w3c.github.io/pointerevents/#pointer-capture
+      if (!requireClicksOriginateLocally) {
+        e.target.releasePointerCapture(e.pointerId);
       }
       // We only care about left mouse.
-      if (e.nativeEvent.button !== LEFT_MOUSE_NUMBER) {
+      // -1 button comes on onPointerEnter so we allow that.
+      if (e.nativeEvent.button !== -1 && e.nativeEvent.button !== LEFT_MOUSE_NUMBER) {
         return;
       }
       setActive(true);
 
-      function onUp(ee: MouseEvent) {
-        if (ee.button !== LEFT_MOUSE_NUMBER) {
-          return;
-        }
+      function onRelease(ee: PointerEvent) {
         setActive(false);
-        document.removeEventListener("mouseup", onUp);
+        document.removeEventListener("pointerup", onRelease);
       }
-      document.addEventListener("mouseup", onUp);
+      document.addEventListener("pointerup", onRelease);
     },
-    [originalOnMouseDown]
+    [originalOnPointerDown]
   );
 
-  const originalOnTouchStart = props.onTouchStart;
-
-  const onTouchStart = useCallback(
-    (e) => {
-      if (originalOnTouchStart != null) {
-        originalOnTouchStart(e);
-      }
-      setActive(true);
-
-      function onUp() {
-        setActive(false);
-        document.removeEventListener("touchend", onUp);
-      }
-      document.addEventListener("touchend", onUp);
-    },
-    [originalOnTouchStart]
-  );
+  // We watch for events onPointerEnter only when requireClicksOriginateLocally === false
+  // If the pointer enters the WinampButton area, and the pointer button is already down, trigger a PointerDown
+  const onPointerEnter = (e: React.PointerEvent<HTMLDivElement>) => {
+    if (e.buttons === 1) {
+      // Emit a pointerup to get the other buttons to release
+      document.dispatchEvent(new Event('pointerup', e));
+      // Simulate a pointerdown on the current button
+      onPointerDown(e);
+    }
+  };
 
+  // Remove the custom prop "requireClicksOriginateLocally" before passing the HTML props to the child div
+  const htmlProps = {...props};
+  delete htmlProps.requireClicksOriginateLocally;
```

Comment:
Neat trick. Can't do this here though. requireClicksOriginateLocally is already declared up in the function params. This will redeclare the same variable as part of the destructuring.

Reply:
Thx. Updated in PR